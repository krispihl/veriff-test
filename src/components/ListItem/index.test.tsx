import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { ListItem } from './index';

describe('ListItem', () => {
    it('renders ListItem component correctly', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: false,
            disabled: false,
        }

        const { container } = render(<ListItem check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('Face is clearly visible', { exact: true})).toBeInTheDocument();
        expect(screen.getByText('Yes', { exact: true})).toBeInTheDocument();
        expect(screen.getByText('No', { exact: true})).toBeInTheDocument();
    });

    it('renders active ListItem correctly', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: true,
            disabled: false,
        }
        const { container } = render(<ListItem check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getByTestId('list-item')).toHaveClass('item--active');
    });

    it('renders disabled ListItem correctly', () => {
        const testCheck = {
            id: 'ccc',
            priority: 7,
            description: 'Face is clearly visible',
            active: false,
            disabled: true,
        }
        const { container } = render(<ListItem check={testCheck} updateChecks={jest.fn()} />);

        expect(container).toMatchSnapshot();
        expect(screen.getByTestId('list-item')).toHaveClass('item--disabled');
    });
})