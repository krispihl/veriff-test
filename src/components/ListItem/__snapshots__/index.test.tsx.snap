// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders ListItem component correctly 1`] = `
<div>
  <div
    class="item"
    data-testid="list-item"
  >
    <p
      class="text"
    >
      Face is clearly visible
    </p>
    <div>
      <button
        class="leftButton"
      >
        Yes
      </button>
      <button
        class="rightButton"
      >
        No
      </button>
    </div>
  </div>
</div>
`;

exports[`ListItem renders active ListItem correctly 1`] = `
<div>
  <div
    class="item item--active"
    data-testid="list-item"
  >
    <p
      class="text"
    >
      Face is clearly visible
    </p>
    <div>
      <button
        class="leftButton"
      >
        Yes
      </button>
      <button
        class="rightButton"
      >
        No
      </button>
    </div>
  </div>
</div>
`;

exports[`ListItem renders disabled ListItem correctly 1`] = `
<div>
  <div
    class="item item--disabled"
    data-testid="list-item"
  >
    <p
      class="text"
    >
      Face is clearly visible
    </p>
    <div>
      <button
        class="leftButton disabled"
      >
        Yes
      </button>
      <button
        class="rightButton disabled"
      >
        No
      </button>
    </div>
  </div>
</div>
`;
